torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -161,6 +161,10 @@ struct TORCH_API AccumulateGrad : public Node {
         // valid operation which adds `new_grad` to `variable_grad` in
         // place. `variable_grad` is thus still referring to the same tensor
         // after the operation.
+        // Also DistributedDataParallel(DDP) package relies on grad is
+        // mutated in place for saving peak memory usage. DDP will still
+        // work correctly if it is mutated out of place here, but it will
+        // increase one copy of grads peak memory usage.
```

Comment:
Not sure this is clear. Do you mean that it will lead to one extra copy (and so slow down) and one extra Tensor to be alive at the same time (and so more memory)?

Reply:
one extra tensor and more memory, I will refactor the comment to make it clear